solutions/swb-reference/integration-tests/tests/multiStep/dataset.test.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+import { v4 as uuidv4 } from 'uuid';
+import { getConstants } from '../../../src/constants';
+import ClientSession from '../../support/clientSession';
+import Setup from '../../support/setup';
+import RandomTextGenerator from '../../support/utils/randomTextGenerator';
+import { uuidRegExp } from '../../support/utils/regExpressions';
+
+describe('multiStep dataset integration test', () => {
+  const setup: Setup = new Setup();
+  let adminSession: ClientSession;
+  let datasetsBucketName: string;
+
+  beforeAll(async () => {
+    adminSession = await setup.getDefaultAdminSession();
+    const aws = setup.getMainAwsClient();
+    const { S3_DATASETS_BUCKET_ARN_OUTPUT_KEY } = getConstants();
+    const {
```

Comment:
When the API deploy happens, a `.js` file is generated in `swb-reference/src/config`. That file has the CFN output of the main stack. Should we read values from that file instead of pulling it from CF? Technically that file should always be present and we can pull values from it.However, that does couple the API test with the API deployment.

Thoughts @SanketD92 @FernandoAranda

Reply:
I thought about decoupling them too, but I guess the integration test framework will eventually depend on the API deployment configs and outputs anyways (like we depend on constants that are linked to the stage config file here). 

You bring up a good point about using the `.js` file, saves a CFN SDK call. I'll make that change.

However this becomes a requirement - running cdk-deploy from the machine before running integration tests. I'll write that note in the test example file.

Reply:
Merging both into integration test config settings actually works best, since we don't need to re-declare some attributes anymore.

Reply:
Will be covering this in a separate PR as discussed.